File: java/src/main/java/gov/doc/ntia/sigmf/Extensions.java
package gov.doc.ntia.sigmf;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import gov.doc.ntia.sigmf.serialization.ExtensionsDeserializer;
import gov.doc.ntia.sigmf.serialization.ExtensionsSerializer;

import java.util.ArrayList;

@JsonSerialize(using = ExtensionsSerializer.class)
@JsonDeserialize(using = ExtensionsDeserializer.class)
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Extensions {
    protected ArrayList<Extension> extensions;

    public Extensions(){
        extensions = new ArrayList<>();
    }

    public ArrayList<Extension> getExtensions() {
        return extensions;
    }

    public void setExtensions(ArrayList<Extension> extensions) {
        this.extensions = extensions;
    }


    public void addExtension(Extension extension) {
        extensions.add(extension);
    }
}

File: Csharp/sigmf/sigmf/Extensions.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace sigmf
{
    [Serializable]
    public class Extensions
    {
        private static readonly long serialVersionUID = 1L;

        protected Extension[] extensions;

        public Extensions()
        {
            extensions = new Extension[400];
        }
        public Extension[] getExtensions()
        {
            return extensions;
        }

        public void setExtensions(Extension[] extensions)
        {
            this.extensions = extensions;
        }

        public int addExtension(Extension extension, int index)
        {
            extensions[index] = extension;
            return index + 1;
        }
    }
}
